core/domain/rights_manager_test.py

```diff
@@ -1052,3 +1157,151 @@ def test_moderator_can_unpublish_published_exploration(self):
         self.assertTrue(rights_manager.check_can_unpublish_activity(
             self.moderator,
             rights_manager.get_exploration_rights(self.published_exp_id)))
+
+
+class ActivityRightsTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(ActivityRightsTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.owner = user_services.UserActionsInfo(self.owner_id)
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+        self.viewer = user_services.UserActionsInfo(self.viewer_id)
+
+        self.exp_id = 'exp_id'
+        self.save_new_valid_exploration(self.exp_id, self.owner_id)
+        rights_manager.publish_exploration(self.owner, self.exp_id)
+        self.activity_rights = rights_manager.get_exploration_rights(
+            self.exp_id)
+
+    def test_validate_community_owned_explorations(self):
+        self.activity_rights.community_owned = True
+        with self.assertRaisesRegexp(
+            Exception,
+            'Community-owned explorations should have no owners, '
+            'editors, voice artists or viewers specified.'):
+            self.activity_rights.validate()
+
+        self.activity_rights.owner_ids = []
+        self.activity_rights.status = rights_manager.ACTIVITY_STATUS_PRIVATE
+        with self.assertRaisesRegexp(
+            Exception, 'Community-owned explorations cannot be private'):
+            self.activity_rights.validate()
+
+    def test_validate_private_explorations(self):
+        self.activity_rights.viewer_ids = [self.viewer_id]
+        with self.assertRaisesRegexp(
+            Exception, 'Public explorations should have no viewers specified.'):
+            self.activity_rights.validate()
+
+    def test_validate_owner_editor(self):
```

Comment:
test_validate_that_owner_cannot_be_editor

Please make all your other test names read more clearly in natural English -- I'm having a lot of trouble deciphering them. Don't leave words out.

Reply:
done